Reduce use of Vector::uncheckedAppend() throughout the code base

cdumez opened the pull request:
#### f39857d5b7716637fa4f3ed71f5aa5df1ea19f7e
<pre>
Reduce use of Vector::uncheckedAppend() throughout the code base
<a href="https://bugs.webkit.org/show_bug.cgi?id=262602">https://bugs.webkit.org/show_bug.cgi?id=262602</a>

Reviewed by Brent Fulgham.

* Source/WTF/wtf/cf/LanguageCF.cpp:
(WTF::platformUserPreferredLanguages):
* Source/WebCore/Modules/ShapeDetection/Implementation/Cocoa/BarcodeDetectorImplementation.mm:
(WebCore::ShapeDetection::BarcodeDetectorImpl::getSupportedFormats):
* Source/WebCore/Modules/WebGPU/GPUQueue.cpp:
(WebCore::GPUQueue::submit):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.cpp:
(WebCore::GPURenderPassEncoder::executeBundles):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp:
(WebCore::WebGPU::DeviceImpl::createBindGroupLayout):
(WebCore::WebGPU::convertToBacking):
* Source/WebCore/Modules/entriesapi/DOMFileSystem.cpp:
(WebCore::listDirectoryWithMetadata):
(WebCore::toFileSystemEntries):
* Source/WebCore/Modules/gamepad/Gamepad.cpp:
* Source/WebCore/Modules/highlight/Highlight.cpp:
(WebCore::Highlight::Highlight):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::toLibWebRTCMediaEndpointTransceiverState):
(WebCore::transceiverStatesFromPeerConnection):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformableFrame.cpp:
(WebCore::LibWebRTCRtpTransformableFrame::audioMetadata const):

Canonical link: <a href="https://commits.webkit.org/268874@main">https://commits.webkit.org/268874@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b691554b49f0a4f7566c707ec91a776718acfcad

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20921 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21995 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22811 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/19492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21159 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24568 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21506 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20767 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21144 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20910 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18167 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23668 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18072 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18981 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25272 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18232 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19157 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19173 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23184 "Passed tests") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/20353 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/19742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/24331 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18991 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5770 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5021 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/25594 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19563 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5596 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/wtf/cf/LanguageCF.cpp:
```diff
@@ -101,12 +101,10 @@ Vector<String> platformUserPreferredLanguages(ShouldMinimizeLanguages shouldMini
         auto platformLanguage = static_cast<CFStringRef>(CFArrayGetValueAtIndex(platformLanguages.get(), i));
-        languages.uncheckedAppend(httpStyleLanguageCode(platformLanguage, shouldMinimizeLanguages));
-    }
+        return httpStyleLanguageCode(platformLanguage, shouldMinimizeLanguages);
```
Clever!

brentfulgham commented on Source/WebCore/Modules/WebGPU/GPUQueue.cpp:
```diff
@@ -54,13 +54,11 @@ void GPUQueue::setLabel(String&& label)
-        result.uncheckedAppend(commandBuffer->backing());
-    }
+    auto result = WTF::compactMap(commandBuffers, [](auto& commandBuffer) -> std::optional<std::reference_wrapper<WebGPU::CommandBuffer>> {
+        if (commandBuffer)
```
Is there a benefit to having the 'work' done in the early return, rather than returning early with `nullopt`? No need to change the code, I'm just curious if there is a benefit to this pattern. It looked strange given that it flips the order of the original code.

cdumez replied:
No benefit. Not sure why I changed the order

brentfulgham commented on Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.cpp:
```diff
@@ -147,13 +147,11 @@ void GPURenderPassEncoder::endOcclusionQuery()
-            continue;
-        result.uncheckedAppend(bundle->backing());
-    }
+    auto result = WTF::compactMap(bundles, [](auto& bundle) -> std::optional<std::reference_wrapper<WebGPU::RenderBundle>> {
```
I had to stop and remind myself that `map` isn't just a data type, it's an algorithm :-D

brentfulgham commented on Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp:
```diff
@@ -187,34 +187,32 @@ Ref<BindGroupLayout> DeviceImpl::createBindGroupLayout(const BindGroupLayoutDesc
+                entry.binding,
+                m_convertToBackingContext->convertShaderStageFlagsToBacking(entry.visibility), {
+                    nullptr,
+                    entry.buffer ? m_convertToBackingContext->convertToBacking(entry.buffer->type) : WGPUBufferBindingType_Undefined,
```
Not your code, but all these ternary operations look funny. I wonder if the constructor could have been templated on entry.buffer fullness.

brentfulgham reviewed (approved):
Great improvement! Even if it weren't a performance gain, the new code is so much clearer to read. r=me

cdumez committed:
Reduce use of Vector::uncheckedAppend() throughout the code base
https://bugs.webkit.org/show_bug.cgi?id=262602

Reviewed by Brent Fulgham.

* Source/WTF/wtf/cf/LanguageCF.cpp:
(WTF::platformUserPreferredLanguages):
* Source/WebCore/Modules/ShapeDetection/Implementation/Cocoa/BarcodeDetectorImplementation.mm:
(WebCore::ShapeDetection::BarcodeDetectorImpl::getSupportedFormats):
* Source/WebCore/Modules/WebGPU/GPUQueue.cpp:
(WebCore::GPUQueue::submit):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.cpp:
(WebCore::GPURenderPassEncoder::executeBundles):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp:
(WebCore::WebGPU::DeviceImpl::createBindGroupLayout):
(WebCore::WebGPU::convertToBacking):
* Source/WebCore/Modules/entriesapi/DOMFileSystem.cpp:
(WebCore::listDirectoryWithMetadata):
(WebCore::toFileSystemEntries):
* Source/WebCore/Modules/gamepad/Gamepad.cpp:
* Source/WebCore/Modules/highlight/Highlight.cpp:
(WebCore::Highlight::Highlight):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::toLibWebRTCMediaEndpointTransceiverState):
(WebCore::transceiverStatesFromPeerConnection):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformableFrame.cpp:
(WebCore::LibWebRTCRtpTransformableFrame::audioMetadata const):

Canonical link: https://commits.webkit.org/268874@main
